File: backend/src/RevisionPlatform.Api/Modules/MultipleChoice/MultipleChoiceContent.cs
namespace RevisionPlatform.Api.Modules.MultipleChoice;

/// <summary>
/// Content of a multiple-choice module: a question, its choices, the ids of the
/// correct choices (one or more) and an optional explanation shown after answering.
/// </summary>
public record MultipleChoiceContent(
    string? Question,
    List<Choice?>? Choices,
    List<string?>? CorrectChoiceIds,
    string? Explanation);

/// <summary>A possible answer. <see cref="Id"/> is unique within the question.</summary>
public record Choice(string? Id, string? Text);

File: frontend/src/app/modules/multiple-choice/multiple-choice-content.ts
/** Content of a multiple-choice module (see docs/api.md). */
export interface MultipleChoiceContent {
  question: string;
  choices: Choice[];
  /** One or more ids of `choices`. */
  correctChoiceIds: string[];
  explanation: string | null;
}

export interface Choice {
  id: string;
  text: string;
}

export const MULTIPLE_CHOICE_LIMITS = {
  questionMaxLength: 1000,
  minChoices: 2,
  maxChoices: 10,
  choiceTextMaxLength: 500,
  explanationMaxLength: 2000,
} as const;
